### DISCUSSION TITLE
SV06 Klipper on the Creality 4.2.7

### DISCUSSION THREAD

**Discussion by LimesKey** [bassamanator/Sovol-SV06-firmware#148] [General]
I recently purchased a Creality 4.2.7 with an STM32F103 microchip from [AliExpress](https://www.aliexpress.com/item/1005004862704098.html) to replace my fried Sovol SV06 motherboard. Because the layout was very similar to both the BTT SKR Mini E3 V3 and the stock Sovol motherboard, I assumed the replacement would be easy and that I would save a few bucks instead of buying either of those. I was mistaken and finding the correct pinouts and Klipper configurations are much harder than I thought. I stumbled upon a full electrical schematic of the board and this did have the pinouts, although difficult to read. I entered all of them into my printer.cfg but I'm of course still having trouble.

After changing the TMC motor drivers to the TMC2208 (TMC2225 are very similar) I get `Option 'uart_address' in section 'tmc2208 stepper_y' must have a maximum of 0` which doesn't make sense to me and after changing it, it still refuses to work. I'm hoping someone more knowledgable than me can spot a mistake I made and it'll be a easy fix.

[Creality.4.2.7.-.Schematic.28-5-22-1.pdf](https://github.com/bassamanator/Sovol-SV06-firmware/files/14304759/Creality.4.2.7.-.Schematic.28-5-22-1.pdf)

[printer.txt](https://github.com/bassamanator/Sovol-SV06-firmware/files/14304767/printer.txt)

**Comment by bassamanator** [comment]
What printer.cfg is that? I looks like mine/SV06 actually.

**Reply by LimesKey** [reply]
It is yours! I started off with yours as the base then changed pinouts and added a few more configs.

**Comment by LimesKey** [comment]
Apparently, it's a limitation with the motors and the shared UART, I think you need to perform with mod which requires soldering to make it work. https://drive.google.com/file/d/1oUj07YvqexBF21gjLlnnvFgyJlPtkhSL/view

I'm just going to return the motherboard and buy a Sovol motherboard
